main/actions/src/main/scala/sbt/ForkTests.scala

```diff
@@ -73,6 +74,8 @@ private[sbt] object ForkTests {
             os.flush()
 
             new React(is, os, log, opts.testListeners, resultsAcc).react()
+          } catch {
+            case NonFatal(e) => resultsAcc("Forked test harness failed: " + e.getMessage) = SuiteResult.Error
```

Comment:
I think we're ok. Here's what happens and the output, using this patch and your test case:

```scala
[error] Uncaught exception when running tests: java.lang.Throwable
[trace] Stack trace suppressed: run last test:test for the full output.
[error] Error: Total 0, Failed 0, Errors 0, Passed 0
[error] Error during tests:
[error]        	Forked test harness failed: null
[error] (test:test) sbt.TestsFailedException: Tests unsuccessful
[error] Total time: 10 s, completed 02-Sep-2016 11:45:25
> last test:test
[debug] javaOptions: List()
[debug] Forking tests - parallelism = false
[debug] Create a single-thread test executor
[error] Uncaught exception when running tests: java.lang.Throwable
sbt.ForkMain$ForkError: java.lang.Throwable: null
       	at build.MyRunner.tasks(MyFramework.scala:14)
       	at sbt.ForkMain$Run.runTests(ForkMain.java:253)
       	at sbt.ForkMain$Run.run(ForkMain.java:139)
       	at sbt.ForkMain.main(ForkMain.java:121)
[debug] Summary for my not available.
[error] Error: Total 0, Failed 0, Errors 0, Passed 0
[error] Error during tests:
[error]        	Forked test harness failed: null
[error] (test:test) sbt.TestsFailedException: Tests unsuccessful
```